polyfill/lib/ecmascript.mjs

```diff
@@ -2782,19 +2746,45 @@ export function FormatDateTimeUTCOffsetRounded(offsetNanoseconds) {
   return FormatOffsetTimeZoneIdentifier(offsetNanoseconds / 60e9);
 }
 
-export function GetUTCEpochNanoseconds(year, month, day, hour, minute, second, millisecond, microsecond, nanosecond) {
+export function GetEpochNanoseconds(
+  year,
+  month,
+  day,
+  hour,
+  minute,
+  second,
+  millisecond,
+  microsecond,
+  nanosecond,
+  offsetNs
+) {
+  let result = GetUTCEpochNanoseconds(year, month, day, hour, minute, second, millisecond, microsecond, nanosecond);
+  if (offsetNs) result = result.subtract(bigInt(offsetNs));
+  ValidateEpochNanoseconds(result);
+  return result;
+}
+
+function GetUTCEpochNanoseconds(year, month, day, hour, minute, second, millisecond, microsecond, nanosecond) {
+  // The pattern of leap years in the ISO 8601 calendar repeats every 400
+  // years. To avoid overflowing at the edges of the range, we reduce the year
+  // to the remainder after dividing by 400, and then add back all the
+  // nanoseconds from the multiples of 400 years at the end.
+  const reducedYear = year % 400;
+  const yearCycles = (year - reducedYear) / 400;
+  const nsIn400YearCycle = bigInt(400 * 365 + 97).multiply(DAY_NANOS);
```

Comment:
This is a constant, and should therefore probably be defined outside the function.